theme/src/components/layout.js

```diff
@@ -27,12 +26,21 @@ function Layout({children, pageContext}) {
     source,
     storybook,
     additionalContributors,
+    componentId,
   } = pageContext.frontmatter
 
   if (!additionalContributors) {
     additionalContributors = []
   }
 
+  const component = componentMetadata.components[componentId]
+
+  // Auto-populate title and description using component metadata
+  if (component) {
+    title ||= component.displayName
```

Comment:
This means a project could override the metadata if they want?

Reply:
Yup. This allows you to override the component metadata using frontmatter variables. So if you provided a `title` in the frontmatter of a markdown file, it will take precidence over the `displayName` stored in the component-metadata repo.